ssh-key/src/sshsig.rs

```diff
@@ -346,3 +349,26 @@ impl Encode for SignedData<'_> {
         Ok(())
     }
 }
+
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for SshSig {
+    fn deserialize<D>(deserializer: D) -> core::result::Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        let string = String::deserialize(deserializer)?;
+        string.parse::<SshSig>().map_err(de::Error::custom)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl Serialize for SshSig {
+    fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        self.to_pem(LineEnding::LF)
```

Comment:
I feel like the implementation should vary depending on 
https://docs.rs/serde/latest/serde/trait.Serializer.html#method.is_human_readable

Format as pem if this is human readable, otherwise just send the wire format as bytes.

Reply:
right! I'll fix that once I get home

Reply:
I've looked into this and it appears there is no function to convert SshSig to bytes, and I don't want to screw anything up while implementing this. I'd recommend someone else add this since it's a high profile library.

Reply:
You can do:

```rust
let mut bytes = Vec::new();
sshsig.encode(&mut bytes)?;
```

Reply:
Ok, also added bytes decode to deserializer :)